File: frmLogin.vb
Imports System.Text
Imports System.Security.Cryptography
Imports System.Threading


Public Class frmLogin

    Dim gds As New gradonDataSet()
    Dim users As DataTable
    Dim gDap As New gradonDataSetTableAdapters.usersTableAdapter()

    Private Sub OK_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnOK.Click

        If txtUser.Text = "" And txtPass.Text = "" Then
            MessageBox.Show("Please complete the fields", "Incomplete", MessageBoxButtons.OK, MessageBoxIcon.Exclamation)
            Exit Sub
        End If

        Dim tMain As New Thread(AddressOf DoLogin)
        tMain.Start()
        DisableControls()

    End Sub

    Private Sub DisableControls()
        For Each con As Control In Me.Controls
            con.Enabled = False
        Next
    End Sub

    Private Sub EnableControls()
        For Each con As Control In Me.Controls
            con.Enabled = True
        Next
    End Sub

    Private Sub DoLogin()
        Dim md5 As String

        md5 = GenerateMD5Hash(txtPass.Text)
        users = gds.Tables("users")
        gDap.Fill(users)

        Dim dr As DataRow() = users.Select("id = '" & txtUser.Text & "' AND password = '" & md5 & "'")

        If (dr.Length = 0) Then
            MessageBox.Show("Invalid ID or password", "Invalid", MessageBoxButtons.OK, MessageBoxIcon.Error)
            Me.Invoke(New MethodInvoker(AddressOf EnableControls))
            Exit Sub
        End If

        Dim type As String = dr(0).Field(Of String)("type")

        Me.Invoke(New MethodInvoker(Sub()

                                        Me.Close()
                                    End Sub))
    End Sub

    Private Sub Cancel_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCancel.Click
        Me.Close()
    End Sub

    Private Function GenerateMD5Hash(ByVal SourceText As String) As String
        Dim md5 As New MD5CryptoServiceProvider
        Dim bytesStr As Byte() = Encoding.ASCII.GetBytes(SourceText)
        Dim out As String = ""

        bytesStr = md5.ComputeHash(bytesStr)

        For Each chr As Byte In bytesStr
            out += chr.ToString("x2")
        Next

        Return out
    End Function

End Class

File: frmLogin.cs
using System.Collections.Generic;
using System;
using System.Linq;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using System.Xml.Linq;
using System.IO;
using System.Configuration;
using System.Collections;
using System.Windows.Forms;
using System.Text;
using System.Security.Cryptography;
using System.Threading;
using MySql.Data.MySqlClient;

namespace Gradon
{
	public partial class frmLogin
	{

        string connStr = System.Configuration.ConfigurationManager.ConnectionStrings["Gradon.My.MySettings.gradonConnectionString"].ConnectionString;
        MySqlConnection mConn;


		public frmLogin()
		{
			InitializeComponent();
			
			//Added to support default instance behavour in C#
			if (defaultInstance == null)
				defaultInstance = this;

            mConn = new MySqlConnection(connStr);
		}
		
#region Default Instance
		
		private static frmLogin defaultInstance;
		
		/// <summary>
		/// Added by the VB.Net to C# Converter to support default instance behavour in C#
		/// </summary>
		public static frmLogin Default
		{
			get
			{
				if (defaultInstance == null)
				{
					defaultInstance = new frmLogin();
					defaultInstance.FormClosed += new FormClosedEventHandler(defaultInstance_FormClosed);
				}
				
				return defaultInstance;
			}
		}
		
		static void defaultInstance_FormClosed(object sender, FormClosedEventArgs e)
		{
			defaultInstance = null;
		}
		
#endregion
		

		//gradonDataSet gds = new gradonDataSet();
		//gradonDataSet.usersDataTable users;
		//gradonDataSetTableAdapters.usersTableAdapter gDap = new gradonDataSetTableAdapters.usersTableAdapter();
		
		public void OK_Click(System.Object sender, System.EventArgs e)
		{
			
			if (txtUser.Text == "" && txtPass.Text == "")
			{
				MessageBox.Show("Please complete the fields", "Incomplete", MessageBoxButtons.OK, MessageBoxIcon.Exclamation);
				return;
			}
			
			Thread tMain = new Thread(new System.Threading.ThreadStart(DoLogin));
			tMain.Start();
			DisableControls();
			
		}
		
		private void DisableControls()
		{
			foreach (Control con in this.Controls)
			{
				con.Enabled = false;
			}
		}
		
		private void EnableControls()
		{
			foreach (Control con in this.Controls)
			{
				con.Enabled = true;
			}
		}
		
		private void DoLogin()
		{
			string md5;
            mConn.Open();
			md5 = GenerateMD5Hash(txtPass.Text);
            MySqlCommand mCmd = new MySqlCommand("SELECT type FROM users WHERE id = \'" + MySqlHelper.EscapeString(txtUser.Text) + "\' AND password = \'" + md5 + "\'", mConn);
            MySqlDataReader mReader = mCmd.ExecuteReader();

			if (!mReader.Read())
			{
				MessageBox.Show("Invalid ID or password", "Invalid", MessageBoxButtons.OK, MessageBoxIcon.Error);
                mConn.Close();
                this.Invoke(new MethodInvoker(EnableControls));
				return;
			}

			string type = mReader.GetString("type");
            mConn.Close();
			this.Invoke(new MethodInvoker(delegate
			{
				switch (type)
				{
					case "admin":
						frmAdmin adFrm = new frmAdmin(txtUser.Text);
						adFrm.Show();
                        break;
                    case "instructor":
                        frmInstructor inFrm = new frmInstructor(txtUser.Text);
                        inFrm.Show();
                        break;
                    case "student":
                        frmStudent fs = new frmStudent(txtUser.Text);
                        fs.Show();
                        break;
				}
				
				this.Close();
			}
			));
		}
		
		public void Cancel_Click(System.Object sender, System.EventArgs e)
		{
			this.Close();
		}
		
		private string GenerateMD5Hash(string SourceText)
		{
			MD5CryptoServiceProvider md5 = new MD5CryptoServiceProvider();
			byte[] bytesStr = Encoding.ASCII.GetBytes(SourceText);
			string @out = "";
			
			bytesStr = md5.ComputeHash(bytesStr);
			
			foreach (byte chr in bytesStr)
			{
				@out += chr.ToString("x2");
			}
			
			return @out;
		}

        private void llReg_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            frmUser fni = new frmUser("new", "student");
            fni.ShowDialog(this);
        }
		
	}
	
}
